pkg/gpu/nvidia/metrics/metrics.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml"
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	// DutyCycle reports the percent of time when the GPU was actively processing.
+	DutyCycle = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "duty_cycle",
+			Help: "Percent of time when the GPU was actively processing",
+		},
+		[]string{"namespace", "pod", "container", "make", "accelerator_id", "model"})
```

Comment:
namespace, pod, container all seem to be high-cardinality fields, and it is against the best practice to store them in labels (https://prometheus.io/docs/practices/naming/#labels). Should we get rid of them?

Reply:
Actually on a single node, we can have at most 16 GPUs. Each GPU can be allocated to only one container at a time. So the cardinality is not that high.

The reason we need these labels is because these metrics are published to the k8s_container resource (https://cloud.google.com/monitoring/api/resources#tag_k8s_container) in stack driver. namespace, pod and container fields will be derived from these labels.